### PR TITLE
chore: remove e2e_slurm_gpu series tests

### PR DESCRIPTION
**Pull request by rb-determined-ai**
Note that there are nightly tests decorated with:

   - @e2e_slurm
   - skipif(not torch.cuda.is_available())

So we still have some GPU-specific slurm tests at this point.  But those tests were not actually running as part of the e2e_slurm_gpu tests anyway.

This is part of a larger effort to get rid of our znode tests, which are notoriously unreliable.

### CODE REVIEW

File: e2e_tests/tests/cluster/test_checkpoints.py [hunk 1]
```diff
@@ -47,7 +47,6 @@ def wait_for_gc_to_finish(sess: api.Session, experiment_ids: List[int]) -> None:
 
 
 @pytest.mark.e2e_gpu
-@pytest.mark.e2e_slurm_gpu
 def test_set_gc_policy() -> None:
```
Comment (rb-determined-ai) [inline e2e_tests/tests/cluster/test_checkpoints.py:50]: wut.  Why was this ever a gpu test.  Seriously.

Instead of demoting this test to e2e_slurm, I moved the e2e_slurm to test_delete_checkpoints which actually tests that checkpoint_gc works.

(I almost deleted `test_set_gc_policy()` last week just because it doesn't check the results of setting gc policy, it only makes sure the CLI doesn't crash(!))

File: e2e_tests/tests/cluster/test_checkpoints.py [hunk 2]
```diff
@@ -121,6 +120,7 @@ def test_gc_checkpoints_lfs() -> None:
 
 
 @pytest.mark.e2e_cpu
+@pytest.mark.e2e_slurm
```
Comment (rb-determined-ai) [inline e2e_tests/tests/cluster/test_checkpoints.py:123]: I'm pretty open to not even testing checkpoint gc on slurm.

afaict the only way it actually fails is if bind mounts breaks, and that seems like we probably test it sufficiently elsewhere.

But I added the test because we don't have a good way to expose the gc failure in general, so having one explicit test seems like the right choice.

File: e2e_tests/tests/experiment/test_profiling.py [hunk 3]
```diff
@@ -12,7 +12,6 @@
 
 
 @pytest.mark.e2e_gpu
-@pytest.mark.e2e_slurm_gpu
```
Comment (rb-determined-ai) [inline e2e_tests/tests/experiment/test_profiling.py:15]: This is testing a rest API, which does not care about the resource manager.

The only thing in python code that could be tested is "can you talk to a GPU", but I think there is basically zero chance that this test fails if gpu training succeeds.

File: e2e_tests/tests/experiment/test_pytorch.py [hunk 4]
```diff
@@ -1,64 +0,0 @@
...
-import pytest
-
-from tests import api_utils
-from tests import config as conf
-from tests import experiment as exp
-
-
-@pytest.mark.parallel
-@pytest.mark.e2e_slurm_gpu
-def test_pytorch_gradient_aggregation() -> None:
```
Comment (rb-determined-ai) [inline e2e_tests/tests/experiment/test_pytorch.py:12]: This test was written before we had parallel gpu tests.  We didn't have a way to validate our gradient aggregation logic (which is ours, not just a library we call), other than an e2e test.

It never needed a slurm test.

This test has been obsoleted by [multi-gpu unit testing](https://github.com/determined-ai/determined/blob/f5400eadc8109ad5ac5885b50475c3df22676211/harness/tests/experiment/pytorch/test_pytorch_trial.py#L980-L982).

File: e2e_tests/tests/nightly/test_pytorch2.py [hunk 5]
```diff
@@ -7,14 +7,13 @@
 
 @pytest.mark.distributed
 @pytest.mark.gpu_required
-@pytest.mark.e2e_slurm_gpu
```
Comment (rb-determined-ai) [inline e2e_tests/tests/nightly/test_pytorch2.py:10]: I can get behind testing our pytorch2 images (so I did not delete the whole test), but I see absolutely no reason why this needs to run on slurm.

File: e2e_tests/tests/nightly/test_pytorch2.py [hunk 6]
```diff
@@ -7,14 +7,13 @@
...
 def test_pytorch2_hf_language_modeling_distributed() -> None:
     sess = api_utils.user_session()
     test_dir = "hf_language_modeling"
 
     config = conf.load_config(conf.hf_trainer_examples_path(f"{test_dir}/distributed.yaml"))
     config = conf.set_pt2_image(config)
-    config = conf.set_slots_per_trial(config, 4)
```
Comment (rb-determined-ai) [inline e2e_tests/tests/nightly/test_pytorch2.py:17]: Bring this gpu test in line with the slots per trial common to our other distributed gpu tests.

File: e2e_tests/tests/cluster/test_checkpoints.py [hunk 7]
```diff
@@ -47,7 +47,6 @@ def wait_for_gc_to_finish(sess: api.Session, experiment_ids: List[int]) -> None:
 
 
 @pytest.mark.e2e_gpu
```
Comment (azhou-determined) [inline e2e_tests/tests/cluster/test_checkpoints.py:49]: does this need to be gpu even?

Comment (azhou-determined) [reply e2e_tests/tests/cluster/test_checkpoints.py:49]: same for `test_gc_checkpoints` here

Comment (azhou-determined) [reply e2e_tests/tests/cluster/test_checkpoints.py:49]: same for `test_s3_no_creds` (though it's being skipped it seems)

Comment (rb-determined-ai) [reply e2e_tests/tests/cluster/test_checkpoints.py:49]: that might be a question for @stoksc, I think maybe that's to run those tests on fewer clusters?

**Review by dannysauer — APPROVED** [review]
From an infra perspective, this is fine.  Developers needed for the other parts.

File: e2e_tests/tests/cluster/test_checkpoints.py (hunk 7 above)
Comment (azhou-determined) [reply e2e_tests/tests/cluster/test_checkpoints.py:49]: wouldn't `e2e_cpu` cause it to run on fewer clusters?

**Comment by azhou-determined** [comment]
not necessarily related to this PR, but i'm looking at `cluster/test_logging.py` specifically
`@pytest.mark.e2e_gpu  # Note, e2e_gpu and not gpu_required hits k8s cpu tests.`

kind of a weird mark? why are we running gpu tests if they're not required? seems like at least some of them are what the comment says, using it to actually run k8 CPU tests. since we have `e2e_k8s`, maybe those tests could be replaced with `e2e_k8s` and `e2e_cpu`? definitely `test_k8s_init_containers` but maybe others too.

i guess the question is how many of these tests actually need GPU/need to be tested on specific architectures.

**Review by stoksc — APPROVED** [review]

**Review by molinamelendezj — APPROVED** [review]

**Review by azhou-determined — APPROVED** [review]
🔥
